plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -426,13 +426,19 @@ export class MongoDBInstrumentation extends InstrumentationBase<
     // mongodb is using "tick" when calling a callback, this way the context
     // in final callback (resultHandler) is lost
     const activeContext = context.active();
+    const config = this._config;
     return function patchedEnd(this: {}, ...args: unknown[]) {
       const error = args[0];
       if (error instanceof Error) {
         span.setStatus({
           code: SpanStatusCode.ERROR,
           message: error.message,
         });
+      } else {
+        const results = args[1] as mongodb.CommandCursorResult;
+        if (config.responseHook !== undefined) {
+          config.responseHook(span, results);
```

Comment:
Run it in a secure manner, use "safeExecuteInTheMiddle"

Reply:
sure - i haven't seen any usage of it in the mongodb instrumentation implementation 
i'll add it here

Reply:
Added